src/resolve/resolvconf-compat.c

```diff
@@ -256,19 +257,28 @@ int resolvconf_parse_argv(int argc, char *argv[]) {
                         log_syntax(NULL, LOG_DEBUG, "stdin", n, 0, "Ignoring resolv.conf line: %s", line);
                 }
 
-                if (type == TYPE_EXCLUSIVE) {
+                switch (type) {
+                case TYPE_REGULAR:
+                        break;
+
+                case TYPE_PRIVATE:
+                        arg_set_default_route = false;
```

Comment:
wouldn't it make sense to explicitly set this field to "true" in the other two cases of this switch statement?